### PR TITLE
feat: add missing contributing docs

### LINKED ISSUE #11: Feature: add missing contributing docs
### Is your feature request related to a problem? Please describe.

Fast forwarding [docs](https://docs.opensauced.pizza) deployment we skipped at least one essential article like "how to add a new document in the docs endpoint".

### Describe the solution you'd like

Potentially have a contributor who's inexperienced with [docusaurus](https://docusaurus.io/docs) add this document portraying their experience and we improve upon it.

### Describe alternatives you've considered

Link people to [docusaurus](https://docusaurus.io/docs) and forcing them to become masters of all.

### PR DESCRIPTION
**Pull request by 0-vortex**
## What type of PR is this? (check all applicable)

- [] ♻️ Refactor
- [ ] ✨ Feature
- [x] 🐛 Bug Fix
- [x] 👷 Optimization
- [x] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This PR adds a lot of documentation on open sauced tooling and some minor bug fixes that would have taken too long to merge if we went the triage way - sorry folks, you missed out on opening more hacktoberfest issues!

## Related Tickets & Documents

Closes #11 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [x] 📜 contributing.md
- [x] 📓 docs
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

Praise the 🍕

### CODE REVIEW

**Review by bdougie — APPROVED** [review]
LGTM, thanks for the check-engine docs.

File: .github/workflows/development.yml [hunk 1]
```diff
@@ -24,8 +24,8 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-node-
 
-      - name: "🔧 install npm@7"
-        run: npm i -g npm@7
+      - name: "🔧 install npm@latest"
+        run: npm i -g npm@latest
```
Comment (bdougie) [inline .github/workflows/development.yml:28]: bold, but I think this development is a great place to test this out.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM node:16-alpine as development
+FROM node:14-alpine as development
```
Comment (bdougie) [inline Dockerfile:1]: Why the drop down to 14 here?

Comment (0-vortex) [reply Dockerfile:1]: 14 is the new 12 - or another way of being confused at times :D

File: .github/workflows/development.yml (hunk 1 above)
Comment (0-vortex) [reply .github/workflows/development.yml:28]: the big issue with most people was upgrading 6->7, from here on I don't think it's an issue if we keep to the latest and catch bugs - however, am planning to abstract the workflows and make them reusable from open-sauced/open-sauced

**Pull request merged**